scr/main/java/bittercode/model/Book.java

```diff
@@ -0,0 +1,65 @@
+package com.bittercode.model;
+
+import java.io.Serializable;
+
+public class Book implements Serializable {
+
+    private String barcode;
+    private String name;
+    private String author;
+    private double price;
+    private int quantity;
+
+    public Book(String barcode, String name, String author, double price, int quantity) {
+        this.barcode = barcode;
+        this.name = name;
+        this.author = author;
+        this.setPrice(price);
+        this.quantity = quantity;
+    }
+
+    public Book() {
+        super();
+    }
+
+    public String getBarcode() {
+        return barcode;
+    }
+
+    public void setBarcode(String barcode) {
+        this.barcode = barcode;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getAuthor() {
+        return author;
+    }
+
+    public void setAuthor(String author) {
+        this.author = author;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding quantity validation check</b></div>
<div id="fix">

Consider adding validation in `setQuantity()` to ensure quantity is not negative. This could prevent invalid inventory states.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -53,3 +53,6 @@
     public void setQuantity(int quantity) {
 +        if (quantity < 0) {
 +            throw new IllegalArgumentException("Quantity cannot be negative");
 +        }
         this.quantity = quantity;
     }
```


```suggestion
   public void setQuantity(int quantity) {
        if (quantity < 0) {
            throw new IllegalArgumentException("Quantity cannot be negative");
        }
       this.quantity = quantity;
   }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2527653709>#eefa44</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged